src/providers/index.ts

```diff
@@ -18,6 +18,7 @@ export interface CommonProviderOptions {
   id: string
   name: string
   type: ProviderType
+  allowDangerousEmailAccountLinking?: boolean
```

Comment:
If I understand correctly that this is only a use case for the OAuth providers, then this config should be in `packages/next-auth/src/providers/oauth.ts`